code/glTF2Asset.inl

```diff
@@ -187,47 +194,75 @@ inline void LazyDict<T>::DetachFromDocument()
 }
 
 template<class T>
+Ref<T> LazyDict<T>::Retrieve(unsigned int i)
+{
+
+    typename Dict::iterator it = mObjsByOIndex.find(i);
+    if (it != mObjsByOIndex.end()) {// already created?
+        return Ref<T>(mObjs, it->second);
+    }
+
+    // read it from the JSON object
+    if (!mDict) {
+        throw DeadlyImportError("GLTF: Missing section \"" + std::string(mDictId) + "\"");
+    }
+
+    if (!mDict->IsArray()) {
+        throw DeadlyImportError("GLTF: Field is not an array \"" + std::string(mDictId) + "\"");
+    }
+
+    Value& obj = mDict->operator[](i);
+
+    if (!obj.IsObject()) {
+        throw DeadlyImportError("GLTF: Object at index \"" + std::to_string(i) + "\" is not a JSON object");
+    }
+
+    T* inst = new T();
+    inst->id = std::string(mDictId) + "_" + std::to_string(i);
+    inst->oIndex = i;
+    ReadMember(obj, "name", inst->name);
+    inst->Read(obj, mAsset);
+
+    return Add(inst);
+}
+
+template<class T>
 Ref<T> LazyDict<T>::Get(unsigned int i)
 {
+
     return Ref<T>(mObjs, i);
+
 }
 
 template<class T>
 Ref<T> LazyDict<T>::Get(const char* id)
 {
     id = T::TranslateId(mAsset, id);
 
-    typename Dict::iterator it = mObjsById.find(id);
+    typename IdDict::iterator it = mObjsById.find(id);
     if (it != mObjsById.end()) { // already created?
         return Ref<T>(mObjs, it->second);
     }
 
-    // read it from the JSON object
-    if (!mDict) {
-        throw DeadlyImportError("GLTF: Missing section \"" + std::string(mDictId) + "\"");
-    }
+    return Create(id);
```

Comment:
Should this Create a new Resource? or should it throw? (as `::Get(unsigned int i)` will throw if the access is out-of-bounds). I think the latter

Reply:
Rather than throw, you could also return an empty `Ref` (`return Ref<T>();`), which might also allow you to get rid of the `Has` member function, since you could do something like:
```c++
if (Ref<T> ref = thing.Get(id)) {
    doSomethingWith(ref);
}
```
Does that make sense?

Reply:
Totally makes sense. Was trying to do exactly that for too long on Sunday, but couldn't get the syntax right.

Reply:
Gotcha. In case you're curious, the reason we can treat a `Ref` as a `bool`, as in the above `if` statement, is because there's an `operator bool()` member function defined on the class: https://github.com/Matter-and-Form/assimp/blob/26c3bbfc0ffcae368c5eb9229ee67c35875a6065/code/glTF2Asset.h#L360. It also overloads the `->` operator, so you can treat it like a pointer.